tests/unit/services/key-manager-test.js

```diff
@@ -605,3 +606,73 @@ test('warning is triggered if registering a macro with modifiers and keyup', asy
   assert.ok(warnings, 'warning is present');
   assert.equal(warnings[0], MODIFIERS_ON_KEYUP_WARNING, 'correct warning was sent');
 });
+
+test('event propagation', async function(assert) {
+	assert.expect(8);
+
+	const macroAttrs = {
+		callback: function() {
+			firstMacroCallCount += 1;
+		},
+		executionKey: 'Control',
+		keyEvent: 'keydown',
+        priority: 1
+	};
+
+	const service = this.subject();
+
+	service.addMacro(macroAttrs);
+
+	assert.equal(get(service, 'keydownMacros').length, 1, 'expected number of keydown macros');
+
+	let macroEvent = {
+		type: 'keydown',
+		key: 'Control'
+	};
+	await dispatchEvent(document.body, macroEvent);
+
+	assert.equal(firstMacroCallCount, 1, 'expected macro callback');
+
+	// add a new container to test events propagation
+	const element = document.createElement('div');
+	element.setAttribute('class', 'grid');
+	document.body.appendChild(element);
+
+	// define a listener
+	const Grid = EmberObject.extend({
+        eventType: null,
+        eventKey: null,
+
+		init() {
+			const grid = $('.grid');
```

Comment:
We would like to see this PR merge. 
I believe we just need a call to `_super` inside the `init` hook, ie `this._super(...arguments)`

Reply:
When I fix the syntax issues and remove the `isPresent` fix, this test doesn't fail i.e., this test is not testing the change proposed in this PR. I'll iterate and push this evening or tomorrow.

Reply:
ah my mistake, thanks @patrickberkeley

Reply:
No worries, I'm still looking into this.